### PR TITLE
[WIP] Horizon: fix unit tests and refactor

### PR DESCRIPTION
**Pull request by Maikol**

### CODE REVIEW

File: packages/horizon/contracts/data-service/GraphDirectory.sol [hunk 1]
```diff
@@ -16,6 +16,7 @@ import { IGraphProxyAdmin } from "@graphprotocol/contracts/contracts/upgrades/IG
 
 import { ICuration } from "@graphprotocol/contracts/contracts/curation/ICuration.sol";
 
+import "forge-std/console.sol";
```
Comment (tmigone) [inline packages/horizon/contracts/data-service/GraphDirectory.sol:19]: 👀

File: packages/horizon/contracts/payments/GraphPayments.sol [hunk 2]
```diff
@@ -69,7 +69,11 @@ contract GraphPayments is Multicall, GraphDirectory, GraphPaymentsStorageV1Stora
         }
 
         // Pay the rest to the receiver
-        uint256 tokensReceiverRemaining = tokens - tokensProtocol - tokensDataService - tokensDelegationPool;
+        uint256 totalCut = tokensProtocol + tokensDataService + tokensDelegationPool;
+        if (totalCut > tokens) {
+            revert GraphPaymentsCollectorInsufficientAmount(tokens, totalCut);
```
Comment (tmigone) [inline packages/horizon/contracts/payments/GraphPayments.sol:74]: Not sure I would use the term `Collector` here, how about just:
```suggestion
            revert GraphPaymentsInsufficientTokens(tokens, totalCut);
```

File: packages/horizon/contracts/payments/PaymentsEscrow.sol [hunk 3]
```diff
@@ -78,14 +78,19 @@ contract PaymentsEscrow is Multicall, GraphDirectory, IPaymentsEscrow {
             revert GraphEscrowThawingPeriodTooLong(withdrawEscrowThawingPeriod, MAX_THAWING_PERIOD);
         }
 
-        revokeCollectorThawingPeriod = revokeCollectorThawingPeriod;
-        withdrawEscrowThawingPeriod = withdrawEscrowThawingPeriod;
+        REVOKE_COLLECTOR_THAWING_PERIOD = revokeCollectorThawingPeriod;
```
Comment (tmigone) [inline packages/horizon/contracts/payments/PaymentsEscrow.sol:81]: LOL very nice catch

**Review by tmigone — COMMENTED** [review]
💪 great one, left a minor comment

File: packages/horizon/contracts/payments/GraphPayments.sol (hunk 2 above)
Comment (Maikol) [reply packages/horizon/contracts/payments/GraphPayments.sol:74]: totally!

File: packages/horizon/contracts/payments/PaymentsEscrow.sol (hunk 3 above)
Comment (Maikol) [reply packages/horizon/contracts/payments/PaymentsEscrow.sol:81]: haha I was so confused for a moment

**Review by tmigone — APPROVED** [review]

**Pull request merged**